spec/controllers/devices_controller_spec.rb

```diff
@@ -0,0 +1,167 @@
+require 'rails_helper'
+
+RSpec.describe DevicesController, type: :controller do
+<<<<<<< HEAD
+<<<<<<< HEAD
+describe 'GET #index' do
+    it 'returns http success' do
+      get :index
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the :index view' do
+      get :index
+      response.should render_template :index
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid attributes' do
+      it 'creates new device' do
+        expect {
+          post :create, device: FactoryGirl.create(:house), house: FactoryGirl.attributes_for(:device)
+        }.to change(House, :count).by(1)
+      end
+      it 'redirects to  house_page' do
+        post :create, house: FactoryGirl.create(:house), device: FactoryGirl.attributes_for(:device)
+        response. should redirect_to houses_url
+      end
+    end
+  end
+
+  describe 'GET #show' do
+    it 'assigns the requested house' do
+      house = FactoryGirl.create(:user)
+      device = FactoryGirl.create(:device)
+
+      get :show, id: device
+      assigns(:device).should eq(device)
+=======
+=======
+
+>>>>>>> devices_backend
+  before(:each) do
+    @house = FactoryGirl.create(:house)
```

Comment:
Usually before blocks are used for stubs and complex objects initialization. Simple objects you can create using `let` like this:
```
let(:house)  { create(:house) }
let(:device) { create(:device)}
let(:log)      { create(:json_log_devices) }
```